consensus/enclave/impl/src/lib.rs

```diff
@@ -2340,6 +2341,330 @@ mod tests {
+                token_id: *token_id1,
+                signer_set: minters_signer_set,
+                mint_limit: 10000,
+            }],
+            nonce,
+            tombstone_block: 2,
+            total_mint_limit: 10000,
+        };
+        let message = mint_config_tx_prefix.hash();
+
+        // Signed with all governors
+        let valid_mint_config_tx = {
+            let signature = MultiSig::new(vec![
+                governor_1.sign(message.as_ref()),
+                governor_2.sign(message.as_ref()),
+                governor_3.sign(message.as_ref()),
+            ]);
+            MintConfigTx {
+                prefix: mint_config_tx_prefix.clone(),
+                signature,
+            }
+        };
+
+        // Missing one governor signature.
+        let invalid_mint_config_tx = {
+            let signature = MultiSig::new(vec![
+                governor_1.sign(message.as_ref()),
+                governor_2.sign(message.as_ref()),
+            ]);
+            MintConfigTx {
+                prefix: mint_config_tx_prefix.clone(),
+                signature,
+            }
+        };
+
+        let recipient1 = AccountKey::random(&mut rng);
+
+        // Signed without one of the needed minters
+        let invalid_mint_tx = create_mint_tx_to_recipient(
+            token_id1,
+            &[
+                Ed25519Pair::from(minter_1.private_key()),
+                Ed25519Pair::from(minter_2.private_key()),
+            ],
+            12,
+            &recipient1.default_subaddress(),
+            &mut rng,
+        );
+
+        // Signed with all minters
+        let valid_mint_tx = create_mint_tx_to_recipient(
+            token_id1,
+            &[minter_1, minter_2, minter_3],
+            12,
+            &recipient1.default_subaddress(),
+            &mut rng,
+        );
+
+        let enclave = SgxConsensusEnclave::new(logger.clone());
+        let blockchain_config = BlockchainConfig {
+            block_version,
+            governors_map: governors_map.clone(),
+            governors_signature: sign_governors_map(&governors_map),
+            ..Default::default()
+        };
+        enclave
+            .enclave_init(
+                &Default::default(),
+                &Default::default(),
+                &None,
+                blockchain_config,
+            )
+            .unwrap();
+
+        // Initialize a ledger.
+        let sender = AccountKey::random(&mut rng);
+        let mut ledger = create_ledger();
+
+        // We want the next block that gets appended to the ledger to exceed the
+        // tombstone limit of the mint config tx, since we want to make sure
+        // that minting that relies on an old MintConfigTx (one that is past
+        // its tombstone block) still validate and mint successfully.
+        let n_blocks = mint_config_tx_prefix.tombstone_block;
+        initialize_ledger(block_version, &mut ledger, n_blocks, &sender, &mut rng);
```

Comment:
:wrench: similar to above, consider if this is check is needed in this test
```suggestion
        initialize_ledger(block_version, &mut ledger, 1, &sender, &mut rng);
```